src/Controller/WebPageArchiveRunController.php

```diff
@@ -0,0 +1,156 @@
+ */
+class WebPageArchiveRunController extends ControllerBase implements ContainerInjectionInterface {
+
+  /**
+   * Displays a Web page archive run  revision.
+   *
+   * @param int $web_page_archive_run_revision
+   *   The Web page archive run  revision ID.
+   *
+   * @return array
+   *   An array suitable for drupal_render().
+   */
+  public function revisionShow($web_page_archive_run_revision) {
+    $web_page_archive_run = $this->entityManager()->getStorage('web_page_archive_run')->loadRevision($web_page_archive_run_revision);
+    $view_builder = $this->entityManager()->getViewBuilder('web_page_archive_run');
+
+    return $view_builder->view($web_page_archive_run);
+  }
+
+  /**
+   * Page title callback for a Web page archive run  revision.
+   *
+   * @param int $web_page_archive_run_revision
+   *   The Web page archive run  revision ID.
+   *
+   * @return string
+   *   The page title.
+   */
+  public function revisionPageTitle($web_page_archive_run_revision) {
+    $web_page_archive_run = $this->entityManager()->getStorage('web_page_archive_run')->loadRevision($web_page_archive_run_revision);
+    return $this->t('Revision of %title from %date', ['%title' => $web_page_archive_run->label(), '%date' => format_date($web_page_archive_run->getRevisionCreationTime())]);
+  }
+
+  /**
+   * Generates an overview table of older revisions of a Web page archive run .
+   *
+   * @param \Drupal\web_page_archive\Entity\WebPageArchiveRunInterface $web_page_archive_run
+   *   A Web page archive run  object.
+   *
+   * @return array
+   *   An array as expected by drupal_render().
+   */
+  public function revisionOverview(WebPageArchiveRunInterface $web_page_archive_run) {
+    $account = $this->currentUser();
+    $langcode = $web_page_archive_run->language()->getId();
+    $langname = $web_page_archive_run->language()->getName();
+    $languages = $web_page_archive_run->getTranslationLanguages();
+    $has_translations = (count($languages) > 1);
+    $web_page_archive_run_storage = $this->entityManager()->getStorage('web_page_archive_run');
+
+    $build['#title'] = $has_translations ? $this->t('@langname revisions for %title', ['@langname' => $langname, '%title' => $web_page_archive_run->label()]) : $this->t('Revisions for %title', ['%title' => $web_page_archive_run->label()]);
+    $header = [$this->t('Revision'), $this->t('Operations')];
+
+    $delete_permission = (($account->hasPermission("delete all web page archive run revisions") || $account->hasPermission('administer web page archive run entities')));
+
+    $rows = [];
+
+    $vids = $web_page_archive_run_storage->revisionIds($web_page_archive_run);
+
+    $latest_revision = TRUE;
+
+    foreach (array_reverse($vids) as $vid) {
+      /** @var \Drupal\web_page_archive\WebPageArchiveRunInterface $revision */
+      $revision = $web_page_archive_run_storage->loadRevision($vid);
+      // Only show revisions that are affected by the language that is being
```

Comment:
I meant that the comment was an artifact from elsewhere, such as Drupal Console. The reference to languages for untranslatable entities is funny, but not an issue for me.